### DISCUSSION TITLE
Websocket implementation

### DISCUSSION THREAD

**Discussion by csaldiasdev** [ergo-services/ergo#160] [Q&A] [answered]
Is possible make a websocket implementation over ergo gen TCP handler?

Or, is better establish a websocket connection over gen WebHandler?

I would like know your opinions :)

**Comment by halturin** [comment, marked as answer]
Thanks for the question. I would consider using websocket library and implementing it on top of the gen.Server (similar to gen.Web).

PS: https://twitter.com/halturin/status/1632707844658135042. We will start to work on it after the release 3.0 happen (scheduled this fall)

**Comment by csaldiasdev** [comment]
Thanks!
I'll waiting any updates about that 3.0 release.
Great project !!!

_But, I'll try make a "native websocket over ergo tcp" only for learning purposes anyway._

**Comment by Kwinin** [comment]
GD job  bro, I will also be very looking forward to  this feature

**Comment by halturin** [comment]
This feature is done in the 3.0 version, so stay tuned :)

**Comment by Northward712** [comment]
I also need this function,I have done my own implementation through WebHandle (based on WebHandle and then websocket.upgrade to a new process custom loop), but I still look forward to the update.
